store/postgres/src/relational_queries.rs

```diff
@@ -884,24 +887,56 @@ impl<'a> fmt::Display for QueryFilter<'a> {
 }
 
 impl<'a> QueryFilter<'a> {
-    pub fn new(filter: &'a EntityFilter, table: &'a Table) -> Result<Self, StoreError> {
-        Self::valid_attributes(filter, table)?;
-        Ok(QueryFilter { filter, table })
+    pub fn new(
+        filter: &'a EntityFilter,
+        table: &'a Table,
+        layout: &'a Layout,
+        block: BlockNumber,
+    ) -> Result<Self, StoreError> {
+        Self::valid_attributes(filter, table, layout, false)?;
+
+        Ok(QueryFilter {
+            filter,
+            table,
+            layout,
+            block,
+            table_prefix: "c.",
+        })
     }
 
-    fn valid_attributes(filter: &'a EntityFilter, table: &'a Table) -> Result<(), StoreError> {
+    fn valid_attributes(
+        filter: &'a EntityFilter,
+        table: &'a Table,
+        layout: &'a Layout,
+        child_filter_ancestor: bool,
+    ) -> Result<(), StoreError> {
         use EntityFilter::*;
         match filter {
             And(filters) | Or(filters) => {
                 for filter in filters {
-                    Self::valid_attributes(filter, table)?;
+                    Self::valid_attributes(filter, table, layout, child_filter_ancestor)?;
                 }
             }
+            Child(attr, entity, child_filter, _) => {
+                if child_filter_ancestor {
+                    return Err(StoreError::QueryExecutionError(
+                        "Only a single level sub filter is allowed".to_string(),
```

Comment:
How about something like `Child filters can not be nested`?

Reply:
f00526fefdf012e559f5a06220aad0fa1d968f86